pkg/csi/service/common/util.go

```diff
@@ -146,23 +146,28 @@ func validateVolumeCapabilities(volCaps []*csi.VolumeCapability, validAccessMode
 			}
 		}
 		if !found {
-			return false
+			return fmt.Errorf("%s access mode is not supported for %q volumes", csi.VolumeCapability_AccessMode_Mode_name[int32(volCap.AccessMode.GetMode())], volumeType)
 		}
 		if volCap.AccessMode.Mode == csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER {
 			if volCap.GetMount() != nil && (volCap.GetMount().FsType == NfsV4FsType || volCap.GetMount().FsType == NfsFsType) {
-				return false
+				return fmt.Errorf("NFS fstype not supported for ReadWriteOnce volume creation")
+			}
+		}
+		if volCap.AccessMode.Mode == csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER {
```

Comment:
Looks like this if-statement is a repeat of previous one. Can you remove one of them?

Reply:
My bad. This is a mistake.

Thanks for catching
I have removed it